qiskit/circuit/__init__.py

```diff
@@ -10,38 +10,62 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""
-========================================
-Quantum Circuits (:mod:`qiskit.circuit`)
-========================================
+r"""
+=============================================
+Quantum circuit model (:mod:`qiskit.circuit`)
+=============================================
 
 .. currentmodule:: qiskit.circuit
 
-Overview
-========
+The fundamental element of quantum computing is the *quantum circuit*.  This is a computational
+routine that can be run, one shot at a time, on a quantum processing unit (QPU).  A circuit will act
+on a predefined amount of quantum data (in Qiskit, we only directly support qubits) with unitary
+operations (gates), measurements and resets.  In addition, a quantum circuit can contain operations
+on classical data, including near-time computations and control-flow constructs, which are executed
+by the controllers of the QPU.
+
+.. note::
+
+    You may wish to skip the introductory material and jump directly to:
+
+    * :ref:`the API overview of the whole circuit module <circuit-module-api>`
+    * :ref:`the detailed discussion about how circuits are represented <circuit-repr>`
+    * the core :class:`QuantumCircuit` class for how to build and query circuits
+    * :ref:`information on construction custom instructions <circuit-custom-gates>`
```

Comment:
```suggestion
    * :ref:`information on the construction of custom operations <circuit-custom-gates>`
```
We haven't talked about instructions, yet. Should we rephrase quantum/classical operations as instructions?

Reply:
This link is specifically about the `Instruction` class, not `Operation`. The reason I didn't use the class name as a Sphinx cross-ref directly is just because it's really tricky/annoying to nest inline syntax elements in rST.

Reply:
Also, about "not defined yet", that's deliberate - this is a quick set of links for people already familiar with Qiskit and all the introductory content, like people who are coming back to the page for a second time. It's like a table of contents - I think it's ok to use this terminology in it.

Reply:
Alright, makes sense to me. :-)